libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Maxell.cpp

```diff
@@ -30,6 +30,45 @@ uint8_t maxell_cell_ids[] = { 0x3f,  // cell 1
  * #define BATTMONITOR_SMBUS_MAXELL_PF_STATUS             0x53    // safety status
 */
 
+/**
+ * detect if a Maxell battery is connected. We'll detect by
+ * trying to take a reading on I2C. If we get a result the battery is
+ * there.
+ *
+ * @param [in] mon
+ * @param [in] mon_state
+ * @param [in] params
+ * @param [in] dev
+ * @return sensor instance
+ */
+AP_BattMonitor_Backend *AP_BattMonitor_SMBus_Maxell::detect(AP_BattMonitor &mon,
+                                                            AP_BattMonitor::BattMonitor_State &mon_state,
+                                                            AP_BattMonitor_Params &params,
+                                                            AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev)
+{
+    AP_BattMonitor_SMBus_Maxell *sensor = new AP_BattMonitor_SMBus_Maxell(mon, mon_state, params, std::move(dev));
+    if (!sensor) {
+        return nullptr;
+    }
+
+    if (sensor->_dev->get_semaphore()->take(HAL_SEMAPHORE_BLOCK_FOREVER)) {
+        uint8_t manufactureName[2] {};
+        if (!sensor->_dev->read_registers(BATTMONITOR_SMBUS_MANUFACTURE_NAME, manufactureName, 2)) {
+            sensor->_dev->get_semaphore()->give();
+            delete sensor;
+            return nullptr;
+        }
+
+        sensor->_dev->get_semaphore()->give();
+        if (manufactureName[0] == 0 || manufactureName[0] > 32 || (manufactureName[1] != 'H' && manufactureName[1] != 'm' && manufactureName[1] != 'M')) {
```

Comment:
Can't bring this part in. This is our generic backend SMBus implementation at this point, and this would block others who are using it.

Reply:
I totally agree of course.. I wonder if simply checking if the I2C request works or doesn't work is sufficient for the detection?  I think all battery manufacturers implement this "manufacturer Name" column.. and we could ask @Pedals2Paddles to test for us on the solo battery.

Reply:
It wouldn't affect solo since it has it's own driver. Most probably implement it, but they aren't strictly required to. (And implementations vary so much). It's probably fairly reasonable to expect it to be filled in, but if we aren't going to sanity check it somehow, we may as well just ask for voltage, since that is one of the fields we require the battery to give us.